### DISCUSSION TITLE
How can I assign a filename to a code block?

### DISCUSSION THREAD

**Discussion by geiregjo** [eslint/markdown#300] [Q&A] [answered]
When the processor extracts code blocks from .md files, each block is individually linted. If I understand correctly, the extracted code block inherits the filename of the .md file it came from.

I need a way to control the filenames assigned to these code blocks. The problem is that I’m using the [CheckFile plugin](https://github.com/dukeluo/eslint-plugin-check-file) for validating filenames. When a README.md file contains Vue code blocks, it results in filename errors because README is not considered a valid Vue filename.

Is there a way to specify a filename annotation for these code blocks? Alternatively, can the plugin be configured with a map of valid filenames and extensions to use?

**Comment by nzakas** [comment, marked as answer]
Right now there is no way to do this. There is a proposal to allow configuring the filename from code blocks: https://github.com/eslint/markdown/pull/230
